lib/pretty_text.rb

```diff
@@ -371,30 +371,14 @@ def self.strip_links(string)
     fragment.to_html
   end
 
-  # Given a Nokogiri doc, convert all links to absolute
-  def self.make_all_links_absolute(doc)
-    site_uri = nil
-    doc.css("a").each do |link|
-      href = link["href"].to_s
-      begin
-        uri = URI(href)
-        site_uri ||= URI(Discourse.base_url)
-        link["href"] = "#{site_uri}#{link['href']}" unless uri.host.present?
-      rescue URI::InvalidURIError, URI::InvalidComponentError
-        # leave it
-      end
-    end
-  end
-
   def self.strip_image_wrapping(doc)
     doc.css(".lightbox-wrapper .meta").remove
   end
 
-  def self.format_for_email(html)
-    doc = Nokogiri::HTML.fragment(html)
-    make_all_links_absolute(doc)
-    strip_image_wrapping(doc)
-    doc.to_html
+  def self.format_for_email(html, style: nil)
```

Comment:
We need to format all of the posts properly for the mailing list summary email, and all that logic lives in `Email::Styles`, so I've swapped it out here. I thought this would work the same way as the digest, but since the digest only deals with post excerpts (ie the first paragraph), it sidesteps a lot of the issues with rendering full posts via email. I don't think there's an existing email which fully renders multiple posts at the same time?

I was pleasantly surprised to see that calling it this way actually takes less time without a style applied (so, for all existing calls to `format_for_email`):

```
existing_format_for_email  4.000000   0.320000   4.320000 (  5.531362)
new_format_for_email       3.600000   0.280000   3.880000 (  4.845140)
```
(run with a medium-sized post, 100x)

However, adding the `notification` style (which calls `Email::Styles.format_notification`, and applies all the bells and whistles normally applied to the user notification emails, including the existing mailing list email) slows it down significantly:

```
with_notification_style 28.250000   0.950000  29.200000 ( 32.747481)
```
which is almost 10x slower and .02 seconds per post... seems like a lot if we're sending out emails with heaps of posts in them. But we do need to run it in order to get things like quotes and oneboxes looking decent.

Here's a breakdown of the things that `format_notification` is calling (again, at 100x for a medium-sized post):
```
total                            2.470000   0.090000   2.560000 (  2.670170)
style_times_13                   0.870000   0.020000   0.890000 (  0.887273) # (we call the 'style' method 13 times for different style applications)
correct_first_body_margin        0.070000   0.000000   0.070000 (  0.074445)
correct_footer_style             0.070000   0.000000   0.070000 (  0.071749)
reset_tables                     0.060000   0.000000   0.060000 (  0.060174)
onebox_styles                    0.920000   0.030000   0.950000 (  0.940647)
plugin_styles                    0.440000   0.030000   0.470000 (  0.567873)
```
tl;dr it's mostly the calls to `styles` and onebox.

Thoughts would be to either render the whole html email, and run `format_notification` once on the whole thing (might be quicker?), or have each post store an additional 'email' version of itself in the DB, alongside 'raw' and 'cooked'. Both seem somewhat hard to me, and would rather go with something easier for this PR. Maybe I'm missing something simpler?

Maybe @ZogStriP or perf-master @SamSaffron has some thoughts on this?